api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -137,13 +140,34 @@ public void syncBillingProjectStatus() {
 
   public void cleanBillingBuffer() {
     Iterables.concat(
-            billingProjectBufferEntryDao.findAllByStatusAndLastStatusChangedTimeLessThan(
-                DbStorageEnums.billingProjectBufferStatusToStorage(CREATING),
-                new Timestamp(
-                    clock
-                        .instant()
-                        .minus(CREATING_TIMEOUT_MINUTES, ChronoUnit.MINUTES)
-                        .toEpochMilli())),
+            StreamSupport.stream(
+                    billingProjectBufferEntryDao
+                        .findAllByStatusAndLastStatusChangedTimeLessThan(
+                            DbStorageEnums.billingProjectBufferStatusToStorage(CREATING),
+                            new Timestamp(
+                                clock
+                                    .instant()
+                                    .minus(CREATING_TIMEOUT_MINUTES, ChronoUnit.MINUTES)
+                                    .toEpochMilli()))
+                        .spliterator(),
+                    false)
+                // Ensure that we've also synchronized the status since the deadline elapsed. This
+                // covers degenerate cases where syncBillingProjectStatus might be backlogged and
+                // hasn't caught up to check all the CREATING projects, or covers periods where
+                // the Firecloud API is unresponsive. Ideally we'd push this logic directly down
+                // into the DAO, but JPQL has poor support for date comparisons. Keep the above
+                // filter as well as a first pass filter to limit what we pull into memory.
+                .filter(
+                    entry ->
+                        entry.getLastSyncRequestTime() != null
```

Comment:
nit: I'd do `Optional.ofNullable(entry.getLastSyncRequestTime()).map(...).flatMap(Streams::stream)` (to squash out the empty optionals). I suppose this is primarily a style preference though, but I feel like null checks are a bit of a smell in stream logic.

Better yet, I'd likely just make a helper function like `isBufferEntryInTimespan()` and filter on that using a method reference. This would also make the filter function(s) separately testable.